### ISSUE TITLE
Steam crashes Linux Mint to login screen

### ISSUE THREAD

**Issue by The10axe** [ValveSoftware/steam-for-linux#8555]
#### Your [system information](https://gist.github.com/The10axe/31d307ab94a91786d63139e7f9b87c1c)

* Steam client version (build number or date): 14/03/2022 at 19:49:17
* Distribution (e.g. Ubuntu): Linux Mint Cinnamon
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
When I launch Steam, it crashes Cinnamon to login screen.

#### Steps for reproducing this issue:

1. Launch steam normally
2. Crashes to login screen

#### Workaround n°1:

1. Launch Steam from taskbar and select things like "Store", "Library".

#### Workaround n°2:

1. Launch Steam via console.

**Comment by kisak-valve** [comment]
Hello @The10axe, in general, no OpenGL application including Steam should be capable of crashing the user's session, and this is most likely a video driver issue.

**Comment by The10axe** [comment]
> Hello @The10axe, in general, no OpenGL application including Steam should be capable of crashing the user's session, and this is most likely a video driver issue.

I'd agree with you, however, it is weird that starting Steam through command prompt or running it by trying to open the Store straight away wouldn't crash the user's session but launching it straight from the app menu or desktop icon would. And I don't understand why the video driver would like the same app some way but not others.

**Comment by lhriley** [comment]
> I'd agree with you, however, it is weird that starting Steam through command prompt or running it by trying to open the Store straight away wouldn't crash the user's session but launching it straight from the app menu or desktop icon would. And I don't understand why the video driver would like the same app some way but not others.

I can verify that launching via a terminal seems to be fine, but launching from the Cinnamon app menu entry crashes to desktop.

The menu item appears to launch this command on my system:
```
sh -c 'STEAM_FRAME_FORCE_CLOSE=1 steam' %U
```

**Comment by kisak-valve** [comment]
Hello @lhriley, would you happen to have an Optimus laptop or multiple GPUs?

**Comment by lhriley** [comment]
@kisak-valve 
```
❯ inxi -G
Graphics:  Device-1: Intel UHD Graphics 630 driver: i915 v: kernel 
           Device-2: NVIDIA TU117GLM [Quadro T1000 Mobile] driver: nvidia v: 510.73.05 
           Display: x11 server: X.Org 1.20.13 driver: modesetting,nvidia unloaded: fbdev,nouveau,vesa tty: N/A 
           OpenGL: renderer: Quadro T1000/PCIe/SSE2 v: 4.6.0 NVIDIA 510.73.05 
❯ uname -a
Linux peitho 5.15.0-33-generic #34~20.04.1-Ubuntu SMP Thu May 19 15:51:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by kisak-valve** [comment]
I'm guessing that `OpenGL: renderer: Quadro T1000/PCIe/SSE2 v: 4.6.0 NVIDIA 510.73.05` means that X is running on the NVIDIA gpu? If that's true, then #8179 may be relevant.

**Comment by lhriley** [comment]
Yes, I have the hybrid graphics enabled, but Optimus is running in `performance` mode / using the discrete GPU.

**Comment by kisak-valve** [comment]
It might be worth experimenting with a shortcut pointing to something else and have `PrefersNonDefaultGPU` enabled with it, maybe chromium, and then reporting your findings to your video driver vendors.

**Comment by lhriley** [comment]
Tested against these without any issues:

```
/usr/share/applications/xed.desktop (text editor)
/usr/share/applications/code.desktop (vscode)

```

**Comment by daxpryce** [comment]
I'm on Mint 20.3 and just started having this same problem.  I can confirm that editing `/usr/share/applications/steam.desktop` and commenting out:

```
PrefersNonDefaultGPU=true
X-KDE-RunOnDiscreteGpu=true
```

Allows steam to start again.

I will note that I installed steam and used it for a few days without a computer restart, and it was only after I restarted for the first time after installing (and running) steam that this started happening.  I thought for sure it was a problem with my Index until I unplugged it and rebooted and kept running into similar problems.
